cmd/state/internal/cmdtree/cmdtree.go

```diff
@@ -295,7 +295,12 @@ func newStateCommand(globals *globalOptions, prime *primer.Values) *captain.Comm
 				Shorthand:   "o",
 				Description: locale.T("flag_state_output_description"),
 				Persist:     true,
-				Value:       &globals.Output,
+				OnUse: func() {
+					if prime.Output().Type().IsStructured() {
+						globals.NonInteractive = true
+					}
```

Comment:
This function is called after our outputer is updated to use the type given on the command line, so we know at this time if it's structured (e.g. json) or not (e.g. plain).